modules/core/src/test/java/org/apache/ignite/internal/processors/datastreamer/DataStreamerImplSelfTest.java

```diff
@@ -110,6 +115,246 @@
         return cfg;
     }
 
+    /**
+     * @throws Exception If failed.
+     */
+    public void testSimilarFuturePerBatch() throws Exception {
+        cnt = 0;
+
+        startGrids(2);
+
+        IgniteDataStreamer<Integer, Integer> dataLdr = grid(0).dataStreamer(DEFAULT_CACHE_NAME);
+
+        IgniteFuture curFut;
+
+        IgniteFuture prevFut = dataLdr.addData(1, 1);
+
+        for (int i = 2; i <= DATA_AMOUNT; i++) {
+            curFut = dataLdr.addData(i, i);
+
+            if (i % IgniteDataStreamer.DFLT_PER_THREAD_BUFFER_SIZE == 1)
```

Comment:
1. Let's rewrite this if into two loops.
2. Let's set explicit value to `petThreadBufSize`, otherwise tests become broken after refactoring(DFLT_PER_THREAD_BUFFER_SIZE bigger then DATA_AMOUNT)